kernel/src/snapshot.rs

```diff
@@ -229,6 +230,13 @@ impl Snapshot {
     pub fn protocol(&self) -> &Protocol {
         &self.protocol
     }
+
+    pub fn column_mapping_mode(&self) -> DeltaResult<ColumnMappingMode> {
```

Comment:
Given that we always access it, recommend to make this a field rather than a method, similar to `metadata` and `protocol`. That way, it can be validated at snapshot creation time, and we no longer need to pass it as a separate arg to the various functions that need it. I think it would also remove the need to make scan builder return a `Result` (tho maybe for future-proofing sake we should still do that)?

Reply:
Yep, done.

`get_state_info` still returns a `Result` because actually pulling the physical name out of the field could fail, so `build` needs to return a `Result`

Also, as you say, I imagine in the future things could go wrong in `build` as well, so I think keeping it as a result makes sense.